pkg/redirectpolicy/manager.go

```diff
@@ -499,6 +509,16 @@ func (rpm *Manager) upsertService(config *LRPConfig, frontendMapping *feMapping)
 	}
 }
 
+func (rpm *Manager) checkPodReady(pod *slimcorev1.Pod) bool {
+	ready := slimcorev1.ConditionUnknown
+	for _, cond := range pod.Status.Conditions {
+		if cond.Type == slimcorev1.PodReady {
+			ready = cond.Status
```

Comment:
Is there supposed to be a `break` after hitting this?

Reply:
If I'm reading this right, it will set `ready` when the pod's condition is `PodReady`, but it continues looping.

Reply:
you are absolutely right, fixed.